src/main/java/ch/njol/skript/effects/EffPlaySound.java

```diff
@@ -47,186 +46,129 @@
-			if (pitchNumber == null)
-				return;
-			pitch = pitchNumber.floatValue();
-		}
+	protected void execute(Event event) {
+		SoundCategory category = this.category == null ? SoundCategory.MASTER : this.category.getOptionalSingle(event)
+			.orElse(SoundCategory.MASTER);
+		float volume = this.volume.getOptionalSingle(event)
+			.orElse(1)
+			.floatValue();
+		float pitch = this.pitch.getOptionalSingle(event)
+			.orElse(1)
+			.floatValue();
+		
 		if (players != null) {
 			if (locations == null) {
-				for (Player p : players.getArray(e))
-					playSound(p, p.getLocation(), sounds.getArray(e), (SoundCategory) category,  volume, pitch);
+				for (Player player : players.getArray(event))
+					SoundReceiver.play(Player::playSound, Player::playSound, player,
+						player.getLocation(), sounds.getArray(event), category,  volume, pitch);
 			} else {
-				for (Player p : players.getArray(e)) {
-					for (Location location : locations.getArray(e))
-						playSound(p, location, sounds.getArray(e), (SoundCategory) category, volume, pitch);
+				for (Player player : players.getArray(event)) {
+					for (Location location : locations.getArray(event))
+						SoundReceiver.play(Player::playSound, Player::playSound, player,
+							location, sounds.getArray(event), category, volume, pitch);
 				}
 			}
-		} else {
-			if (locations != null) {
-				for (Location location : locations.getArray(e))
-					playSound(location, sounds.getArray(e), (SoundCategory) category, volume, pitch);
-			}
+		} else if (locations != null) {
+			for (Location location : locations.getArray(event))
+				SoundReceiver.play(World::playSound, World::playSound, location.getWorld(),
+					location, sounds.getArray(event), category, volume, pitch);
 		}
 	}
 
-	private static void playSound(Player p, Location location, String[] sounds, SoundCategory category, float volume, float pitch) {
-		for (String sound : sounds) {
-			Sound soundEnum = null;
-			try {
-				soundEnum = Sound.valueOf(sound.toUpperCase(Locale.ENGLISH));
-			} catch (IllegalArgumentException ignored) {}
-			if (SOUND_CATEGORIES_EXIST) {
-				if (soundEnum == null) {
-					sound = sound.toLowerCase(Locale.ENGLISH);
-					if (!SOUND_VALID_PATTERN.matcher(sound).matches())
-						continue;
-					p.playSound(location, sound, category, volume, pitch);
-				} else {
-					p.playSound(location, soundEnum, category, volume, pitch);
-				}
-			} else {
-				if (soundEnum == null) {
-					sound = sound.toLowerCase(Locale.ENGLISH);
-					if (!SOUND_VALID_PATTERN.matcher(sound).matches())
-						continue;
-					p.playSound(location, sound, volume, pitch);
-				} else {
-					p.playSound(location, soundEnum, volume, pitch);
-				}
-			}
-		}
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		StringBuilder builder = new StringBuilder()
+			.append("play sound ")
+			.append(sounds.toString(event, debug));
+		
+		if (category != null) builder.append(" in ")
```

Comment:
Inline if statements are not allowed due to Skript's code convention (though currently it doesn't mention that, probably it was missed)